### PR TITLE
Make content_for raise an exception in case of silently overwriting content

### PR DESCRIPTION
**Pull request by Fjan**
This addresses #704

### CODE REVIEW

**Comment by ddfreyne** [comment]
Could you add tests for this, and remove the commented-out code?

**Comment by Fjan** [comment]
OK, done

File: test/helpers/test_capturing.rb [hunk 1]
```diff
@@ -101,6 +101,32 @@ def test_different_sites
...
   end
 
+  def test_content_for_duplicate_symbols
+    with_site do |_site|
+      # Prepare
+      File.open('lib/helpers.rb', 'w') do |io|
+        io.write 'include Nanoc::Helpers::Capturing'
+      end
+      File.open('content/includer.erb', 'w') do |io|
+        io.write '[<%= content_for(@items.find { |i| i.identifier == \'/includee/\' }, :blah) %>]'
```
Comment (ddfreyne) [inline test/helpers/test_capturing.rb]: Replace

```ruby
@items.find { |i| i.identifier == \'/includee/\' }
```

with

```ruby
@items['/includee/']
end

File: lib/nanoc/helpers/capturing.rb [hunk 2]
```diff
@@ -33,7 +33,12 @@ def initialize
 
       def []=(item, name, content)
         @store[item.identifier] ||= {}
-        @store[item.identifier][name] = content
+        if @store[item.identifier][name]
+          return if @store[item.identifier][name] == content
+          raise KeyError, "Content_for was called twice with the same key: #{name}, this would overwrite content for the first call"
```
Comment (ddfreyne) [inline lib/nanoc/helpers/capturing.rb]: I don’t think `KeyError` is correct—that is an exception used when the value of a given key isn’t found. I think raising a string is OK.

File: test/helpers/test_capturing.rb (hunk 1 above)
Comment (Fjan) [inline test/helpers/test_capturing.rb]: Ok, I changed that, but note that all other tests in that file use the other style

**Comment by ddfreyne** [comment]
Thanks!

**Pull request merged**